### PR TITLE
Fix docstrings for `gensim.models.logentropy_model`

### PR DESCRIPTION
**Pull request by minggli**
Motivation **RE:** https://gist.github.com/menshikh-iv/075824fdc4906a3b1b74b8f94ce64b59

This PR improves and corrects the documentations of https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/models/logentropy_model.py

Changes include:
- clarify and separate the behaviours of construct and `initialize`. Existing docs were not as accurate.
- describe parameters of public methods.
- miscellaneous such as docstrings at module level.
- minor styling for imports separating standard library imports.

### CODE REVIEW

File: gensim/models/logentropy_model.py [hunk 1]
```diff
@@ -3,17 +3,18 @@
 #
 # Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
 
+"""This module contains LogEntropyModel class"""
```
Comment (menshikh-iv) [inline gensim/models/logentropy_model.py]: Need to add more description "what is it"
- small description with links
- executable example

File: gensim/models/logentropy_model.py [hunk 2]
```diff
@@ -33,10 +34,11 @@ class LogEntropyModel(interfaces.TransformationABC):
 
     The main methods are:
```
Comment (menshikh-iv) [inline gensim/models/logentropy_model.py]: Example is better than words :)

File: gensim/models/logentropy_model.py [hunk 3]
```diff
@@ -46,9 +48,15 @@ class LogEntropyModel(interfaces.TransformationABC):
...
     def __init__(self, corpus, normalize=True):
-        """
-        `normalize` dictates whether the resulting vectors will be
-        set to unit length.
+        """Constructs an instance of LogEntropyModel, initializes internal statistics of the corpus
+        and calls `initialize` method automatically.
+
+        Parameters
+        ----------
+        corpus : iterable object
```
Comment (menshikh-iv) [inline gensim/models/logentropy_model.py]: `iterable of list of (int, int)`, here and everywhere

File: gensim/models/logentropy_model.py [hunk 4]
```diff
@@ -46,9 +48,15 @@ class LogEntropyModel(interfaces.TransformationABC):
...
+            itertable consisting of word-documents that are made up of term id and term frequency.
+        normalize : bool, optional
+            `normalize` dictates whether the resulting vectors will be set to unit length.
```
Comment (menshikh-iv) [inline gensim/models/logentropy_model.py]: `normalize` -> `parameter`, also it will be good if you present both cases

`if True, ..., otherwise ....`

**Comment by minggli** [comment]
Changes implemented.

File: gensim/models/logentropy_model.py (hunk 4 above)
Comment (minggli) [reply gensim/models/logentropy_model.py]: done

**Comment by menshikh-iv** [comment]
Great @minggli, I'll look into on next week, please wait for a little.

**Comment by menshikh-iv** [comment]
Need to resolve #1807 first (because we have formulas in this PR that should be rendered with `latex`)

**Comment by menshikh-iv** [comment]
Thank you @minggli, congratz with the first contribution to gensim :1st_place_medal:

**Comment by minggli** [comment]
pleasure. finally giving something back to the community.

**Pull request merged**